.github/workflows/codeql.yml

```diff
@@ -0,0 +1,70 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+name: "CodeQL"
+
+on:
```

Comment:
I think it's better to just run this on push and with a schedule first so that we can monitor how good the scanner is before fully committing to run on every PR.  My main concern is that it might be noisy and causes unnecessary frictions